### DISCUSSION TITLE
Custom recipe definitions

### DISCUSSION THREAD

**Discussion by donour** [mlflow/mlflow#8441] [Ideas]
@sunishsheth2009 and MLFlow team:

I'm curious what you guys would recommend for building a custom recipe. I need to build a custom template that is based on the regression recipe, but has different steps. For example, I don't want to use a static train/test split but I want to use cross validation.

It seems like the best thing to do is modify the dag because the logic doesn't fit the existing steps. However, creating a new recipe isn't possible without modifying the codebase and redeploying mlflow.

**Comment by SlapDrone** [comment]
+1 this would be super useful, anything deviating from the existing classification / regression templates is lacking in the documentation ATM which makes it hard to understand how to build your own recipe
